README.md

```diff
@@ -162,3 +162,216 @@ ARGS:
     <pattern>    the search pattern, a regular expression (optional)
     <path>       the root directory for the filesystem search (optional)
 ```
+
+## Examples
+
+First to get `fd`'s help run: 
+
+```
+fd --help
+```
+
+Let's assume we have some files we need to search through like so: 
+
+```
+fd_examples
+├── .gitignore
+├── not_file
+├── sub_dir
+│   ├── .here_be_tests
+│   ├── more_dir
+│   │   ├── .not_here
+│   │   ├── even_further_down
+│   │   │   ├── not_me.sh
+│   │   │   ├── test_seven
+│   │   │   └── testing_eight
+│   │   ├── not_file -> /Users/fd_user/fd_examples/not_file
+│   │   └── test_file_six
+│   ├── test_file_five
+│   ├── test_file_four
+│   └── test_file_three
+├── test_file_one
+├── test_file_two
+├── test_one
+└── this_is_a_test
+```
+
+Let's do a recursive search for anything that has the name test in it (`fd` will start in the current directory and follow symbolic links by default):
+
+`fd test`
+
+This will return us: 
+```
+sub_dir/more_dir/even_further_down/test_seven
+sub_dir/more_dir/even_further_down/testing_eight
+sub_dir/more_dir/test_file_six
+sub_dir/test_file_five
+sub_dir/test_file_three
+sub_dir/test_four
+test_file_one
+test_file_two
+test_one
+this_is_a_test
+```
+
+As you can see this matched every file that had 'test' as a substring in it. Whoops! We forgot a hidden file (`.here_be_tests`) Lets get that by telling `fd` we want to see all files:
+
+`fd --hidden 'test'`
```

Comment:
Not 100% sure about this, but my feeling is that we should advertise the short options (`-H` instead of `--hidden`) in these simple examples, since this is what most people would type at the command line, if they use `fd` a lot. The text before the "code" example could still say something like "We can use the `-H` (or `--hidden`) option to search all hidden files and directories:".

In my opinion, the long options are mostly for shell scripts where the longer options serve as documentation.